docs/migrate-to-nativecurlclient.md

```diff
@@ -0,0 +1,255 @@
+
+This example shows how you can parse the response body of a GET request.
+
+```diff
+-// We dont know if we will get json, xml or a string
+-$dataAsJsonOrXmlOrString = $this->client->get($path);
++$this->client->requestGet($path);
++// $body contains the raw http body of the response
++$body = $this->client->getLastResponseBody();
++
++// if response is XML, create a SimpleXMLElement object
++if ($body !== '' && 0 === strpos($this->client->getLastResponseContentType(), 'application/xml')) {
++    $dataAsXML = new \SimpleXMLElement($body);
++} else if ($body !== '' && 0 === strpos($this->client->getLastResponseContentType(), 'application/json')) {
++    try {
++        $dataAsJson = json_decode($body, true, 512, \JSON_THROW_ON_ERROR);
++    } catch (\JsonException $e) {
++        throw new \Exception('Error decoding body as JSON: '.$e->getMessage());
++    }
++} else {
++    $dataAsString = $body;
++}
+```
+
+### post() to requestPost()
+
+This example shows how you can parse the response body of a POST request.
+
+```diff
+-// We dont know if we will get xml or a string
+-$dataAsXmlOrString = $this->client->post($path, $data);
++$this->client->requestPost($path, $data);
++// $body contains the raw http body of the response
++$body = $this->client->getLastResponseBody();
++
++// if response is XML, create a SimpleXMLElement object
++if ($body !== '' && 0 === strpos($this->client->getLastResponseContentType(), 'application/xml')) {
++    $dataAsXML = new \SimpleXMLElement($body);
++} else {
++    $dataAsString = $body;
++}
+```
+
+### put() to requestPut()
+
+This example shows how you can parse the response body of a PUT request.
+
+```diff
+-// We dont know if we will get xml or a string
+-$dataAsXmlOrString = $this->client->put($path, $data);
++$this->client->requestPut($path, $data);
++// $body contains the raw http body of the response
++$body = $this->client->getLastResponseBody();
++
++// if response is XML, create a SimpleXMLElement object
++if ($body !== '' && 0 === strpos($this->client->getLastResponseContentType(), 'application/xml')) {
++    $dataAsXML = new \SimpleXMLElement($body);
++} else {
++    $dataAsString = $body;
++}
+```
+
+### delete() to requestDelete()
+
+This example shows how you can parse the response body of a DELETE request.
+
+```diff
+-$dataAsString = $this->client->delete($path);
++$this->client->requestDelte($path);
++$dataAsString = $this->client->getLastResponseBody();
+```
+
+### setImpersonateUser() to startImpersonateUser()
+
+If you are using the [Redmine user impersonation](https://www.redmine.org/projects/redmine/wiki/Rest_api#User-Impersonation)
+you have to change your code.
+
+```diff
+// impersonate the user `robin`
+-$client->setImpersonateUser('robin');
++$client->startImpersonateUser('robin');
+
+$userData = $client->getApi('user')->getCurrentUser();
+
+// Now stop impersonation
+-$client->setImpersonateUser(null);
++$client->stopImpersonateUser();
```

Comment:
```suggestion
+$client->stopUserImpersonation();
```

Reply:
Same here: `stopImpersonateUser()` is already part of the released client interface.